src/traces/treemap/attributes.js

```diff
@@ -0,0 +1,273 @@
+            editType: 'plot',
+            description: [
+                'Determines if the positions obtained from solver are flipped on each axis.'
+            ].join(' ')
+        },
+
+        pad: {
+            valType: 'number',
+            role: 'style',
+            min: 0,
+            dflt: 3,
+            editType: 'plot',
+            description: [
+                'Sets the inner padding (in px).'
+            ].join(' ')
+        },
+
+        editType: 'calc',
+    },
+
+    marker: extendFlat({
+        pad: {
+            t: {
+                valType: 'number',
+                role: 'style',
+                min: 0,
+                editType: 'plot',
+                description: [
+                    'Sets the padding form the top (in px).'
+                ].join(' ')
+            },
+            l: {
+                valType: 'number',
+                role: 'style',
+                min: 0,
+                editType: 'plot',
+                description: [
+                    'Sets the padding form the left (in px).'
+                ].join(' ')
+            },
+            r: {
+                valType: 'number',
+                role: 'style',
+                min: 0,
+                editType: 'plot',
+                description: [
+                    'Sets the padding form the right (in px).'
+                ].join(' ')
+            },
+            b: {
+                valType: 'number',
+                role: 'style',
+                min: 0,
+                editType: 'plot',
+                description: [
+                    'Sets the padding form the bottom (in px).'
+                ].join(' ')
+            },
+
+            editType: 'calc'
+        },
+
+        colors: sunburstAttrs.marker.colors,
+
+        opacity: {
+            valType: 'number',
+            editType: 'style',
+            role: 'style',
+            min: 0,
+            max: 1,
+            description: [
+                'Sets the opacity of the branches i.e. based on the hierarchy height.',
+                'With colorscale it is defaulted to 1; otherwise it is defaulted to 0.7'
+            ].join(' ')
+        },
+
+        line: sunburstAttrs.marker.line,
+
+        editType: 'calc'
+    },
+        colorScaleAttrs('marker', {
+            colorAttr: 'colors',
+            anim: false // TODO: set to anim: true?
+        })
+    ),
+
+    leaf: {
+        opacity: {
+            valType: 'number',
+            editType: 'style',
+            role: 'style',
+            min: 0,
+            max: 1,
+            dflt: 1,
+            description: [
+                'Sets the opacity of the leaves.'
+            ].join(' ')
+        },
+        editType: 'plot'
+    },
+
+    pathbar: {
+        visible: {
+            valType: 'boolean',
+            dflt: true,
+            role: 'info',
+            editType: 'plot',
+            description: [
+                'Determines if the path bar is drawn.'
```

Comment:
Post 4e3b7f3744f3cc016e8cdffcf59cd01ee76aabe1, we should probably say something about the pathbar being drawn outside the trace domain.

Reply:
Good call. Done in https://github.com/plotly/plotly.js/pull/4185/commits/a54ca9cf84cdd159df832a67c36c77e387488593.

Reply:
Thanks!